torch/csrc/utils/tensor_new.cpp

```diff
@@ -211,10 +212,15 @@ Tensor internal_new_from_data(const Type & type, at::optional<Device> device_opt
                                                        *torch::getLayout(type.backend()),
                                                        type_inference_device_type);
       const auto& type_to_use = type_inference ? type_inference_type : type;
-      return copy_variables ? new_with_tensor_copy(type_to_use, var, device_index) :
+      // temporarily mute requires_grad when copy from var with requires_grad = false
+      if (mute_grad) var.set_requires_grad(false);
+      auto new_tensor = copy_variables ? new_with_tensor_copy(type_to_use, var, device_index) :
                               new_with_type_conversion(type_to_use, var, device_index);
+      if (mute_grad) var.set_requires_grad(true);
+      return new_tensor;
```

Comment:
Can't you set requires_grad = true on the new_tensor? that would avoid concurrency issues because no other threads hav access to the new tensor yet

Reply:
I guess you meant to set `requires_grad=false` on the new_tensor. Yes, I can do that if it is sufficient. Please note that `grad_fn` will still be created for new_tensor at [s_copy_](https://github.com/pytorch/pytorch/blob/9a7c196040f55d921a4deee42cbb1dd294b8f23b/tools/autograd/templates/VariableType.cpp#L408-L415). If we need to be more careful, then I can also pass in a `requires_grad` args to `s_copy_` to avoid silently ignored `requires_grad` at `torch.tensor()`. Which way is preferred? @ezyang @apaszke